plugins/modules/web_infrastructure/bugzilla.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Chuang Cao <chcao@redhat.com>
+# Atlassian open-source approval reference OSR-76.
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+module: bugzilla
+version_added: "2.10"
+short_description: Manage bugs in a Bugzilla instance by xmlrpc
+description:
+  - This module can be used to fetch, comment, change status of bugzilla ids in a Bugzilla instance.
+
+options:
+  uri:
+    required: true
+    description:
+      - Base URI for the Bugzilla instance.
+    type: str
+
+  operation:
+    required: true
+    aliases: [ command ]
+    type: str
+    choices: [ comment, transition, fetch ]
+    description:
+      - The operation to perform.
+
+  username:
+    required: true
+    type: str
+    description:
+      - The username to log-in with.
+
+  password:
+    required: true
+    type: str
+    description:
+      - The password to log-in with.
+
+  bug:
+    required: false
+    type: str
+    description:
+     - An existing bug number to operate on.
+
+  comment:
+    required: false
+    type: str
+    description:
+     - The comment text to add.
+
+  status:
+    required: false
+    type: str
+    description:
+     - The desired status; only relevant for the transition operation.
+
+  timeout:
+    required: false
+    type: float
+    description:
+      - Set timeout, in seconds, on requests to Bugzilla API.
+    default: 10
+
+notes:
+  - "Currently this only works with REST API."
+
+author: "Chuang Cao (@cc)"
+"""
+
+EXAMPLES = """
+# Add a comment to bug:
+- name: Comment on bug
```

Comment:
and i'm not familiar with bugzilla but if there are not only bug reports possible there (and the module can work with it), i'd use something more general, like 'issue' instead of 'bug'.

Reply:
When I hear 'issue', it's more like JIRA, so I prefer we keep the 'bug' here.